crates/chain/tests/common/tx_template.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+
+use bdk_chain::{tx_graph::TxGraph, ConfirmationHeightAnchor, SpkTxOutIndex};
+use bitcoin::{
+    hashes::Hash, locktime::absolute::LockTime, secp256k1::Secp256k1, OutPoint, ScriptBuf,
+    Sequence, Transaction, TxIn, TxOut, Txid, Witness,
+};
+use miniscript::Descriptor;
+
+/// Transaction template.
+pub struct TxTemplate<'a, A> {
+    /// Uniquely identifies the transaction, before it can have a txid.
+    pub tx_name: &'a str,
+    pub inputs: &'a [TxInTemplate<'a>],
+    pub outputs: &'a [TxOutTemplate],
+    pub anchors: &'a [A],
+    pub last_seen: Option<u64>,
+}
+
+impl<'a, A> Default for TxTemplate<'a, A> {
+    fn default() -> Self {
+        Self {
+            tx_name: Default::default(),
+            inputs: Default::default(),
+            outputs: Default::default(),
+            anchors: Default::default(),
+            last_seen: Default::default(),
+        }
+    }
+}
+
+impl<'a, A> Copy for TxTemplate<'a, A> {}
+
+impl<'a, A> Clone for TxTemplate<'a, A> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+#[allow(dead_code)]
+pub enum TxInTemplate<'a> {
+    /// This will give a random txid and vout.
+    Bogus,
+
+    /// This is used for coinbase transactions because they do not have previous outputs.
+    Coinbase,
+
+    /// Contains the `tx_name` and `vout` that we are spending. The rule is that we must only spend
+    /// a previous transaction.
+    PrevTx(&'a str, usize),
+}
+
+pub struct TxOutTemplate {
+    pub value: u64,
+    pub spk_index: Option<u32>, // some = get spk from SpkTxOutIndex, none = random spk
```

Comment:
Thanks for the comment! The thinking is that it is meant to be a "template" so it should be easy to read and write out. An index is perfect for this, instead of having the full spk.